src/safeposix/syscalls/net_calls.rs

```diff
@@ -2911,34 +2976,52 @@ impl Cage {
             None => interface::RustDuration::MAX,
         };
 
+        // according to standard, we should clear all revents
+        for structpoll in &mut *fds {
+            structpoll.revents = 0;
+        }
+
+        // we loop until either timeout
+        // or any of the file descriptor is ready
         loop {
+            // iterate through each file descriptor
             for structpoll in &mut *fds {
+                // get the file descriptor
                 let fd = structpoll.fd;
+
+                // according to standard, we should ignore all file descriptor
+                // that is smaller than 0
+                if fd < 0 {
+                    continue;
```

Comment:
Small doubt, here we are continuing if the fd is negative. Can you explain why we do this?

Reply:
this is the behavior specified in standard, so we are just following it. Allowing this behavior has one convenient use that user could simply set the fd to negative to temporarily disable monitoring this file descriptor.